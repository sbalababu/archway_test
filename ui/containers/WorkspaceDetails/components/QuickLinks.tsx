import * as React from 'react';
import { Dropdown, Menu, Button, Icon } from 'antd';

import { WebLocation, HueService, YarnService } from '../../../models/Cluster';
import { HiveAllocation } from '../../../models/Workspace';

interface Props {
  hue?: HueService;
  yarn?: YarnService;
  selectedAllocation?: HiveAllocation;
}

const Link = ({ name, host }: { name: string; host: WebLocation }) => {
  const location = `${host.host}:${host.port}`;
  const hint = `${name} (${host.host.substring(0, host.host.indexOf('.'))})`;
  return (
    <Menu.Item>
      <a style={{ padding: 10 }} key="action" target="_blank" rel="noopener noreferrer" href={`//${location}/`}>
        {hint}
      </a>
    </Menu.Item>
  );
};

const QuickLinks = ({ hue, yarn, selectedAllocation }: Props) => (
  <Dropdown
    overlay={
      <Menu onClick={() => null}>
        {hue && hue.load_balancer.map((l, i) => <Link key={`hue-$i`} name="Hue" host={l} />)}
        {yarn && yarn.resource_manager.map((l, i) => <Link key={`rn-$i`} name="Resource Manager" host={l} />)}
      </Menu>
    }
    trigger={['click']}
  >
    <Button type="primary">
      QUICK LINKS <Icon type="down" />
    </Button>
  </Dropdown>
);

export default QuickLinks;
