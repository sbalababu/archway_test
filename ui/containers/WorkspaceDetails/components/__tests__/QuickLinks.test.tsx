import * as React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import QuickLinks from '../QuickLinks';

describe('QuickLinks', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<QuickLinks />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
