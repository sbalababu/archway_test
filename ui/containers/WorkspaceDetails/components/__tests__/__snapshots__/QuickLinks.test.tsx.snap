// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuickLinks renders correctly 1`] = `
<Dropdown
  mouseEnterDelay={0.15}
  mouseLeaveDelay={0.1}
  overlay={
    <Menu
      onClick={[Function]}
    />
  }
  placement="bottomLeft"
  trigger={
    Array [
      "click",
    ]
  }
>
  <Button
    block={false}
    ghost={false}
    htmlType="button"
    loading={false}
    type="primary"
  >
    QUICK LINKS 
    <Icon
      type="down"
    />
  </Button>
</Dropdown>
`;
